CSharp/JobPrep/Program.cs

```diff
@@ -0,0 +1,333 @@
+                    cloudWorkItem.Commit();
+
+                    //Wait for an active job
+                    TimeSpan maxJobCreationTimeout = TimeSpan.FromSeconds(90);
+                    DateTime jobCreationStartTime = DateTime.Now;
+                    DateTime jobCreationTimeoutTime = jobCreationStartTime.Add(maxJobCreationTimeout);
+
+                    cloudWorkItem = wm.GetWorkItem(workItemName);
+
+                    Console.WriteLine("Waiting for a job to become active...");
+                    while (cloudWorkItem.ExecutionInformation == null || cloudWorkItem.ExecutionInformation.RecentJob == null)
+                    {
+                        cloudWorkItem.Refresh();
+                        if (DateTime.Now > jobCreationTimeoutTime)
+                        {
+                            throw new Exception("Timed out waiting for job.");
+                        }
+                        Thread.Sleep(TimeSpan.FromSeconds(5));
+                    }
+
+                    string jobName = cloudWorkItem.ExecutionInformation.RecentJob.Name;
+                    Console.WriteLine("Found job {0}. Adding task objects...", jobName);
+                    ICloudJob job = wm.GetJob(workItemName, jobName);
+
+                    job.AddTask(new CloudTask("task1", "hostname"));
+                    job.AddTask(new CloudTask("task2", "cmd /c dir /s ..\\.."));
+
+                    bool cont = WaitAllJobPrepTask(
+                        client,
+                        PoolName,
+                        workItemName,
+                        jobName,
+                        (succ, fail) => { return succ >= 2 || fail > 0; }, // stop indicator
+                        (succ, fail) => { return succ >= 2; }, // succeess indicator
+                        timeout: 60*1000,
+                        freq: 5*1000,
+                        parallelism: 10
+                        );
+
+                    if (cont) // got enough nodes, continue
+                    {
+                        //We use the task state monitor to monitor the state of our tasks -- in this case we will wait for them all to complete.
+                        ITaskStateMonitor taskStateMonitor = toolbox.CreateTaskStateMonitor();
+
+                        // blocking wait on the list of tasks until all tasks reach completed state
+                        bool timedOut = taskStateMonitor.WaitAll(job.ListTasks(), TaskState.Completed, new TimeSpan(0, 20, 0));
+
+                        if (timedOut)
+                        {
+                            throw new TimeoutException("Timed out waiting for tasks");
+                        }
+
+                        // dump task output
+                        foreach (var t in job.ListTasks())
+                        {
+                            Console.WriteLine("Task " + t.Name + " says:\n" + t.GetTaskFile(Constants.StandardOutFileName).ReadAsString());
```

Comment:
This might be picky, but can you replace "says" with "output"?
I don't like the idea of Tasks thinking for themselves and saying things :)